ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/cucushift-installer-rehearse-gcp-ipi-workflow.yaml

```diff
@@ -2,12 +2,15 @@ workflow:
   as: cucushift-installer-rehearse-gcp-ipi
   steps: 
     pre:
-    - chain: ipi-conf-gcp
-    - chain: ipi-install
+    - chain: cucushift-installer-rehearse-gcp-ipi-pre
```

Comment:
let us keep consistent naming style, replace `pre/post` with `provision/deprovision`.

Reply:
But you can see "pre" & "post" were used by existing workflows and also shorter words. Why must use "provision" & "deprovision" instead? 

```
$ yq e .workflow.steps ci-operator/step-registry/ipi/azure/ipi-azure-workflow.yaml
pre:
  - chain: ipi-azure-pre
post:
  - chain: ipi-azure-post
$ yq e .workflow.steps ci-operator/step-registry/ipi/gcp/ipi-gcp-workflow.yaml
pre:
  - chain: ipi-gcp-pre
post:
  - chain: ipi-gcp-post
$ yq e .workflow.steps ci-operator/step-registry/ipi/aws/ipi-aws-workflow.yaml
pre:
  - chain: ipi-aws-pre
post:
  - chain: ipi-aws-post
$ yq e .workflow.steps ci-operator/step-registry/ipi/alibabacloud/ipi-alibabacloud-workflow.yaml
pre:
  - chain: ipi-alibabacloud-pre
post:
  - chain: ipi-alibabacloud-post
$

Reply:
1. Just for consistent naming style, aws/azure/vsphere are using provision/deprovision
2. The workflow yaml files are placed in QE's rehearse folder, we can manage the naming format on our own.

From technology perspective, it is not a MUST, mainly for a consistent type from team collaboration.

Reply:
May I suggest using "pre" & "post" as they are shorter worders?

Reply:
Per other platform owners, I did not hear much complaint about the long naming, to be one team, I do not hope gcp platform is a uniq one.

Reply:
But, except the new workflows added by QE, I guess most existing workflows were using the words "pre" & "post" instead, so you cannot say it's a unique one.

Reply:
If that, I do not know how to persuade you, and I also do not understand why gcp platform become a different one among installer QE managed workflow yaml files.

Reply:
1. Under "ci-operator/step-registry/ipi" and "ci-operator/step-registry/upi", there are 30 workflows, and "pre" & "post" were used in 24 workflows. So I think for the sake of consistency, we'd better choose the same words as existing workflows.
2. Besides, ".workflow.steps" does be using "pre:" & "post:".

@jianlinliu With the above said, if you still think we should use "provision" & "deprovision", I will update for GCP. 
@yunjiang29 WDYT

Reply:
the `-provision` and `-deprovision` are the chains that we provide for e2e tests. 
installer team is responsible for creating an `install chain` and a `destroy chain` for sub-teams, so that they can create any cluster in their ci job, just like what we are doing in Flexy and ci-profiles.

> But you can see "pre" & "post" were used by existing workflows and also shorter words. Why must use "provision" & "deprovision" instead?

there is also a chain named `ipi-aws-pre`, pre/test/post are controlled by the ci-operator framework, these are 3 typical steps for the tests, I think chains with the suffix `-pre` or `-post` is only used to identify that this chain will be placed in the pre stage.
But what happened in the `-pre` stage? to make it more clear, we use `-provision` and `-deprovision` to represent the creation and deletion of clusters, and this will be closer to existing flexy/ci-profiles concepts, workflow is clear and easy to read, e.g:

```
workflow:
  as: my-image-registry-e2e-tests
  steps: 
    pre:
    - chain: cucushift-installer-rehearse-aws-ipi-private-provision
    - chain: configure-image-registry
    - ref: pull-particular-images  
    test:
    - chain: run-image-registry-test-cases
    post:
    - ref: remove-s3-bucket
    - chain: cucushift-installer-rehearse-aws-ipi-private-deprovision
```